src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,140 @@
+     *
+     * @return A String as the String representation of the TaskList in specified format.
+     */
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder("");
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            String entry = String.format("%d. %s\n",
+                    i + 1, task.toString());
+            result.append(entry);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Returns the size of the ArrayList contained in the class instance.
+     *
+     * @return An int object indicating the size of the current ArrayList.
+     */
+    public int size() {
+        return taskList.size();
+    }
+
+    /**
+     * Returns a String representation of the ArrayList after being filtered by a LocalDate Object.
+     *
+     * @param dateFilter A LocalDate object used to filter the ArrayList of Task objects.
+     * @return A String representation of the ArrayList after being filtered by a LocalDate Object.
+     */
+    public String listSchedule(LocalDate dateFilter) {
+        ArrayList<Task> filteredTaskList = taskList.stream()
+                .filter(ele -> ele.hasSchedule()
+                        && ele.getTime().equals(dateFilter.format(DateTimeFormatter.ofPattern("MMM dd yyyy"))))
+                .collect(Collectors.toCollection(ArrayList::new));
+        StringBuilder result = new StringBuilder("");
+        for (int i = 0; i < filteredTaskList.size(); i++) {
+            Task task = filteredTaskList.get(i);
+            String entry = String.format("%d. %s \n",
+                    i + 1, task.toString());
+            result.append(entry);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Returns a String representation of the ArrayList after being filtered by a String.
+     * @param searchFilter A String used to filter the ArrayList of Task objects.
+     * @return A String representation of the ArrayList after being filtered by a String.
+     */
+    public String listFind(String searchFilter) {
+        ArrayList<Task> filteredTaskList = this.taskList.stream()
+                .filter(ele -> ele.getContent().contains(searchFilter))
+                .collect(Collectors.toCollection(ArrayList::new));
+        StringBuilder result = new StringBuilder("");
+        for (int i = 0; i < filteredTaskList.size(); i++) {
+            Task task = filteredTaskList.get(i);
+            String entry = String.format("%d. %s \n",
+                    i + 1, task.toString());
+            result.append(entry);
+        }
+        return result.toString();
+    }
+
+    /**
+     * Marks the specified Task in the ArrayList as done.
+     *
+     * @param index An int indicating the position of Task in the ArrayList that is to be marked as done.
+     */
+    public void doneTask(int index) {
+        Task holder = taskList.get(index);
+        holder.doneTask();
+    }
```

Comment:
Should TaskList and Task object have the same method name here? It might be confusing.

Reply:
Hmm a good point from you Jerome so any name you would like to suggest please? Cause it seems regardless what alternative I can currently think of, it will sound pretty similar and still cause the confusion.